### PR TITLE
Dashboard description API

### PR DESCRIPTION
**Pull request by jinhyukchang**
### Summary of Changes

Added:
 - Dashboard description get and put API
 - common description put & get swagger schema
Updated:
 - Table description get & put API to use common description swagger schema
 - Neo4j proxy to have common get & put description method for any resource type to reuse.

### Tests

Unit test added. Integration test has been done.

### Documentation

docstring

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

File: metadata_service/entity/resource_type.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum, auto
+
+
+class ResourceType(Enum):
```
Comment (feng-tao) [inline metadata_service/entity/resource_type.py]: may put a comment on why we don't put user here given user is considered a resource entity in Amundsen.

File: metadata_service/__init__.py [hunk 2]
```diff
@@ -80,7 +81,7 @@ def create_app(*, config_module_class: str) -> Flask:
     api.add_resource(PopularTablesAPI, '/popular_tables/')
     api.add_resource(TableDetailAPI, '/table/<path:table_uri>')
     api.add_resource(TableDescriptionAPI,
-                     '/table/<path:table_uri>/description')
+                     '/table/<path:id>/description')
     api.add_resource(TableTagAPI,
                      '/table/<path:table_uri>/tag/<tag>')
```
[the remark this answers is not in the document]
Comment (jinhyukchang) [reply metadata_service/__init__.py]: > should we change this to id as well to be consistent?

Yeah, I hope so but I would pass it for the next time as it requires integration testing.

File: metadata_service/api/dashboard.py [hunk 3]
```diff
@@ -24,3 +28,41 @@ def get(self, *, id: Optional[str] = None) -> Iterable[Union[Mapping, int, None]
...
             return {'message': 'dashboard_id {} does not exist'.format(id)}, HTTPStatus.NOT_FOUND
+
+
+class DashboardDescriptionAPI(BaseAPI):
+    """
+    DashboardDescriptionAPI supports PUT and GET operation to upsert table description
+    """
+    def __init__(self) -> None:
+        self.client = get_proxy_client()
+        super().__init__(DescriptionSchema, 'dashboard_description', self.client)
```
[the remark this answers is not in the document]
Comment (jinhyukchang) [reply metadata_service/api/dashboard.py]: Unlike Java, it turns out Python does not require it to be first 😄 And super() needs `self.client` in this case.

File: metadata_service/entity/resource_type.py (hunk 1 above)
Comment (jinhyukchang) [reply metadata_service/entity/resource_type.py]: Mmm... Actually, I can put `User` here as well. Will do that.

**Comment by feng-tao** [comment]
ship

**Pull request merged**